frontends/verilog/verilog_parser.y

```diff
@@ -616,7 +616,9 @@ non_opt_delay:
 	'#' '(' expr ')' { delete $3; } |
 	'#' '(' expr ',' expr ')' { delete $3; delete $5; } |
 	'#' '(' expr ',' expr ',' expr ')' { delete $3; delete $5; delete $7; } |
-	'#' '(' expr ':' expr ':' expr ')' { delete $3; delete $5; delete $7; };
+	'#' '(' expr ':' expr ':' expr ')' { delete $3; delete $5; delete $7; } |
+	'#' '(' expr ':' expr ':' expr ',' expr ':' expr ':' expr ')' { delete $3; delete $5; delete $7; delete $9; delete $11; delete $13; } |
+	'#' '(' expr ':' expr ':' expr ',' expr ':' expr ':' expr ',' expr ':' expr ':' expr ')' { delete $3; delete $5; delete $7; delete $9; delete $11; delete $13; delete $15; delete $17; delete $19; };
```

Comment:
My reading of the grammar is that a delay like `#(A, A:B:C, D)` should be allowed. To accomplish this and clean things up a bit, I think it would be worthwhile to split out `mintypmax_expr` into its own rule with productions for `expr` and `expr ':' expr ':' expr` (per the official grammar), and have those productions delete the expressions. Then the non-`delay_value` productions in `non_opt_delay` can be something like:
```
'#' '(' mintypmax_expr ')'
'#' '(' mintypmax_expr ',' mintypmax_expr ')'
'#' '(' mintypmax_expr ',' mintypmax_expr ',' mintypmax_expr ')'
```

Reply:
Thanks for giving me the chance but I'm spread pretty thin, I'll revisit this if you haven't done it...!

Reply:
Is delay_risefall.ys line 168 & 170 legitimate? Interestingly it reveals a case when a min-typ-max expression is accepted but general delay expression is not. I'll send a squash when needed

Reply:
I believe that test case is legal. I believe the LRM would allow for the parentheses to be removed, but Yosys does not currently support this. Adding support in Yosys isn't simple (or necessary for the time being) as it would introduce conflicts in the grammar.

Reply:
OK, I don't have much of a comment (haven't used much Verilog). Seems like we went above & beyond. See you on next project.